_posts/2024-01-16-fido-passkey.md

```diff
@@ -0,0 +1,1127 @@
+---
+layout: post
+authors: [nicholas_meyers]
+title: 'Understanding FIDO & Passkeys: Guide for Spring Boot and Angular Apps'
+image: /img/2024-01-16-fido-passkey/banner.png
+tags: [Security, Spring Boot, Angular]
+category: Security
+comments: true
+---
+
+> What's a common vulnerability in applications? Isn't it the traditional username/password login? Perhaps it's time to embrace FIDO/Passkeys for a more secure login method.
+
+- [Introduction](#introduction)
+- [How does it work](#how-does-it-work)
+- [Create a new Spring Boot application](#create-a-new-spring-boot-application)
+- [Create a new Angular application](#create-a-new-angular-application)
+- [Conclusion](#conclusion)
+- [Extra](#extra)
+
+## Introduction
+
+Every week, we encounter articles detailing stolen user login credentials, hacked databases with compromised usernames and passwords ([Car maintenance company leaks 12.7k US phone numbers, emails and MD5 unsalted passwords](https://cybernews.com/security/xado-leaks-us-phone-numbers-emails-md5-unsalted-passwords/){:target="_blank" rel="noopener noreferrer"}, [COMB: largest breach of all time leaked online with 3.2 billion records](https://cybernews.com/news/largest-compilation-of-emails-and-passwords-leaked-free/){:target="_blank" rel="noopener noreferrer"}). 
+Users often reuse the same password across multiple platforms, fail to rotate passwords, or use weak ones. Have you explored the website '[have i been pwned?](https://haveibeenpwned.com/){:target="_blank" rel="noopener noreferrer"}'? 
+Take a look; you'll be surprised at how many of your passwords have already been leaked.
+
+It appears to be high time to address the vulnerabilities associated with traditional username and password logins and transition towards the use of FIDO.
+
+FIDO (Fast IDentity Online) is a set of open authentication standards that aims to replace traditional passwords with more secure and convenient methods.
+
+Key benefits of FIDO authentication:
+* Strong Security: FIDO Authentication utilises public-key cryptography, which is more secure than traditional passwords against phishing attacks, keyloggers, and other threats.
+* Convenience: Passkeys, the credential type enabled by FIDO Authentication, are stored on your device and can be unlocked using biometrics like fingerprint or face recognition, eliminating the need to remember or type passwords repeatedly.
+
+## How does it work
+
+To understand how passkeys work, let's take a step back and delve into the realm of asymmetric cryptography. In asymmetric cryptography, 
+we deal with a key pair consisting of a private key, which must remain private and undisclosed, and a public key, which can be shared openly.
+
+Through asymmetric encryption, a message encrypted with the public key can only be deciphered by the possessor of the private key. 
+Conversely, if a message is encrypted with the private key, it can be read using the public key (signature).
```

Comment:
```suggestion
Conversely, if a message is encrypted with the private key, it can be decrypted using the public key. 
This method is used in signatures, to validate the identity of the sender of the information.
```